src/tr1/game/objects/general/pickup.c

```diff
@@ -50,15 +59,75 @@ static const OBJECT_BOUNDS m_PickUpBoundsUW = {
     },
 };
 
+static void M_Initialise(int16_t item_num);
+static void M_Control(int16_t item_num);
+static void M_SpawnPickupAid(const ITEM *item);
 static void M_GetItem(int16_t item_num, ITEM *item, ITEM *lara_item);
 static void M_GetAllAtLaraPos(ITEM *item, ITEM *lara_item);
 
+static void M_Initialise(int16_t item_num)
+{
+    ITEM *const item = Item_Get(item_num);
+    item->priv = (void *)(intptr_t)(-1);
+    if (item->status != IS_INVISIBLE) {
+        Item_AddActive(item_num);
+    }
+}
+
+static void M_Control(int16_t item_num)
+{
+    ITEM *const item = Item_Get(item_num);
+    if (item->status == IS_INVISIBLE || item->status == IS_DEACTIVATED) {
+        Item_RemoveActive(item_num);
+        return;
+    }
+
+    const ITEM *const lara = Lara_GetItem();
+    if (!g_Config.gameplay.enable_pickup_aids || item->fall_speed != 0
+        || lara == NULL || !Object_GetObject(O_PICKUP_AID)->loaded) {
+        return;
+    }
+
+    const int32_t distance = Item_GetDistance(lara, &item->pos);
+    if (distance < AID_DIST_MIN || distance > AID_DIST_MAX) {
+        return;
+    }
+
+    int32_t timer = (int32_t)(intptr_t)item->priv;
+    if (timer <= 0 || (timer < AID_WAIT_MIN && (Random_GetDraw() % 7) == 0)) {
```

Comment:
Thanks. My only nitpick is that the more common way to use `Random_GetDraw()` is by doing `Random_GetDraw() < 0x4000`, instead of using the modulo operator like `Random_GetDraw() % 7 == 0`.

Reply:
Thanks, I've updated this to stick to the conventions.